lib/mkl/fft.jl

```diff
@@ -0,0 +1,426 @@
+        ccall_set_int64_array(desc, Int32(DFT_PARAM_BWD_STRIDES), strides)
+    end
+    stc = ccall_commit(desc, q); stc == 0 || error("commit failed ($stc)")
+    return cMKLFFTPlan{T,MKLFFT_INVERSE,false,N,R,Nothing}(desc,q,size(X),size(X),false,reg,nothing,nothing)
+end
+
+# In-place (provide separate methods)
+function plan_fft!(X::oneAPI.oneArray{T,N}, region) where {T<:Union{ComplexF32,ComplexF64},N}
+    R = length(region); reg = NTuple{R,Int}(region)
+    # For now, only support full transforms (all dimensions)
+    if reg != ntuple(identity, N)
+        @info "Partial dimension FFT not yet supported. Region $reg must be $(ntuple(identity, N))"
+    end
+    desc,q = _create_descriptor(size(X),T,true)
+    ccall_set_cfg(desc, Int32(DFT_PARAM_PLACEMENT), Int32(DFT_CFG_INPLACE))
+    if N > 1
+        strides = Vector{Int64}(undef, N+1); strides[1]=0; prod=1
+        @inbounds for i in 1:N
+            strides[i+1]=prod; prod*=size(X,i)
+        end
+        ccall_set_int64_array(desc, Int32(DFT_PARAM_FWD_STRIDES), strides)
+        ccall_set_int64_array(desc, Int32(DFT_PARAM_BWD_STRIDES), strides)
+    end
+    stc = ccall_commit(desc, q); stc == 0 || error("commit failed ($stc)")
+    cMKLFFTPlan{T,MKLFFT_FORWARD,true,N,R,Nothing}(desc,q,size(X),size(X),false,reg,nothing,nothing)
+end
+function plan_bfft!(X::oneAPI.oneArray{T,N}, region) where {T<:Union{ComplexF32,ComplexF64},N}
+    R = length(region); reg = NTuple{R,Int}(region)
+    # For now, only support full transforms (all dimensions)
+    if reg != ntuple(identity, N)
+        @info "Partial dimension FFT not yet supported. Region $reg must be $(ntuple(identity, N))"
+    end
+    desc,q = _create_descriptor(size(X),T,true)
+    ccall_set_cfg(desc, Int32(DFT_PARAM_PLACEMENT), Int32(DFT_CFG_INPLACE))
+    if N > 1
+        strides = Vector{Int64}(undef, N+1); strides[1]=0; prod=1
+        @inbounds for i in 1:N
+            strides[i+1]=prod; prod*=size(X,i)
+        end
+        ccall_set_int64_array(desc, Int32(DFT_PARAM_FWD_STRIDES), strides)
+        ccall_set_int64_array(desc, Int32(DFT_PARAM_BWD_STRIDES), strides)
+    end
+    stc = ccall_commit(desc, q); stc == 0 || error("commit failed ($stc)")
+    cMKLFFTPlan{T,MKLFFT_INVERSE,true,N,R,Nothing}(desc,q,size(X),size(X),false,reg,nothing,nothing)
+end
+
+# Real forward (out-of-place) - only support 1D transforms for now
+function plan_rfft(X::oneAPI.oneArray{T,N}, region) where {T<:Union{Float32,Float64},N}
+    # Convert region to tuple if it's a range
+    if isa(region, AbstractUnitRange)
+        # For real FFTs, if region is 1:ndims(X), treat it as (1,) like FFTW
+        if region == 1:N
+            region = (1,)
+        else
+            region = tuple(region...)
+        end
+    end
+    R = length(region); reg = NTuple{R,Int}(region)
+    # Only support single dimension transforms for now
+    if R != 1
+        error("Multi-dimensional real FFT not yet supported")
```

Comment:
@michel2323 Do we know if it is feature not yet implemented by Intel or something to improve on our side?